### PR TITLE
Kinno 556 call receiver

### PR DESCRIPTION
**Pull request by shayb**
call receiver

### CODE REVIEW

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt [hunk 1]
```diff
@@ -14,18 +16,25 @@ import org.kinecosystem.appstransfer.presenter.AppsTransferPresenter
...
-
     private lateinit var dataGroup: Group
     private lateinit var noDataGroup: Group
     private lateinit var loader: ProgressBar
     private lateinit var list: AppsTransferList
     private var presenter: AppsTransferPresenter? = null
 
-    override fun transferToApp(app: EcosystemApp) {
-         Log.d("####", "#### AppsTransferActivity  start transferToApp")
+    override fun onTransferError() {
```
Comment (berryve) [inline appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt:27]: add an error message and / or exception to the method

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt [hunk 2]
```diff
@@ -14,18 +16,25 @@ import org.kinecosystem.appstransfer.presenter.AppsTransferPresenter
...
+        Log.d("####", "#### show pop up error")
+    }
+
+    override fun startAmountChooserActivity(app: EcosystemApp) {
+        Log.d("####", "#### AppsTransferActivity  start startAmountChooserActivity")
```
Comment (berryve) [inline appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt]: "###"

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/AppsTransferPresenter.kt [hunk 3]
```diff
@@ -37,5 +61,24 @@ class AppsTransferPresenter : BasePresenter<IAppsTransferView>(), IAppsTransferP
...
+                Log.d("AppInfoPresenter", "Operation cancelled, no public address received")
+            }
 
+            override fun onError(error: String) {
+                //TODO error cant parse address
+                view?.onTransferError()
+            }
+
+            override fun onResult(address: String) {
+                repository.storeReceiverAppPublicAddress(address)
```
Comment (berryve) [inline appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/AppsTransferPresenter.kt]: This is not a good way of passing back the result of an operation. There is no real reason to store the receiver public address in shared preferences.

The method `processResponse` should either return the public address or take as a parameter a callback that returns the public address.

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt (hunk 1 above)
Comment (shayb) [reply appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt:27]: will be done other pr

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt (hunk 2 above)
Comment (shayb) [reply appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt]: cool

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/AppsTransferPresenter.kt (hunk 3 above)
Comment (shayb) [reply appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/AppsTransferPresenter.kt]: done

**Pull request merged**